testutil/faker.go

```diff
@@ -0,0 +1,62 @@
+package testutil
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/brianvoe/gofakeit/v7"
+	"github.com/stretchr/testify/require"
+)
+
+// Fake will populate any zero fields in the provided struct with fake data.
+// Non-zero fields will remain unchanged.
+// Usage:
+//
+//	key := Fake(t, faker, database.APIKey{
+//	  TokenName: "keep-my-name",
+//	})
+func Fake[T any](t *testing.T, faker *gofakeit.Faker, seed T) T {
+	t.Helper()
+
+	var tmp T
+	err := faker.Struct(&tmp)
+	require.NoError(t, err, "failed to generate fake data for type %T", tmp)
+
+	mergeZero(&seed, tmp)
+	return seed
+}
+
+// mergeZero merges the fields of src into dst, but only if the field in dst is
+// currently the zero value.
+func mergeZero[T any](dst *T, src T) {
```

Comment:
The compiler uses monomorphization for generics, so this adds code for every combination of `T` that this function gets called with, even though the code doesn't change at all. Maybe just use `any` and throw if the types are somehow different. (they're only called from Fake so there's no chance for them to be different)